utils/create-cluster/create-cluster

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Settings
+REDIS_PATH="../../../redis/src"
+REDISRAFT="${PWD}/../../redisraft.so"
+HOST=127.0.0.1
+PORT=5000
+TIMEOUT=2000
+NODES=3
+LOGLEVEL=notice
+CLUSTER_MODE=yes
+ADDITIONAL_OPTIONS=""
+
+# You may want to put the above config parameters into config.sh in order to
+# override the defaults without modifying this script.
+
+if [ -a config.sh ]
+then
+    source "config.sh"
+fi
+
+start_instance() {
+    local port=$1
+
+    echo "Starting $port"
+    $REDIS_PATH/redis-server \
+        --port $port \
+        --loglevel $LOGLEVEL \
+        --logfile ${port}.log \
+        --dbfilename dump-${port}.rdb \
+        --daemonize yes \
+        --loadmodule ${REDISRAFT} \
+            raft-log-filename=raftlog-${port}.db \
+            addr=${HOST}:${port} \
+            cluster-mode=${CLUSTER_MODE} \
+        ${ADDITIONAL_OPTIONS}
+}
+
+stop_instance() {
+    local port=$1
+
+    echo "Stopping $port"
+    ${REDIS_PATH}/redis-cli -p $port shutdown nosave
+}
+
+# Computed vars
+ENDPORT=$((PORT+NODES))
+
+if [ "$1" == "start" ]
+then
+    while [ $((PORT < ENDPORT)) != "0" ]; do
+        PORT=$((PORT+1))
+        start_instance $PORT
+    done
+    exit 0
+fi
+
+if [ "$1" == "create" ]
+then
+    PORT=$((PORT+1))
+    ${REDIS_PATH}/redis-cli -p $PORT raft.cluster init
+    sleep 1
+
+    p=$PORT
+    while [ $((p < ENDPORT)) != "0" ]; do
+        p=$((p+1))
+        ${REDIS_PATH}/redis-cli -p $p raft.cluster join ${HOST}:${PORT}
+    done
+    exit 0
+fi
+
+if [ "$1" == "stop" ]
+then
+    while [ $((PORT < ENDPORT)) != "0" ]; do
+        PORT=$((PORT+1))
+        stop_instance $PORT
+    done
+    exit 0
+fi
+
+if [ "$1" == "watch" ]
+then
+    if [ "$2" == "" ]; then
+        INSTANCE=1
+    else
+        INSTANCE=$2
+    fi
+
+    PORT=$((PORT+INSTANCE))
+    while [ 1 ]; do
+        clear
+        echo "`date` | $HOST:$PORT | RAFT.INFO"
+        ${REDIS_PATH}/redis-cli -p $PORT --raw raft.info
+        sleep 1
+    done
+    exit 0
+fi
+
+if [ "$1" == "tail" ]
+then
+    INSTANCE=$2
+    PORT=$((PORT+INSTANCE))
+    tail -f ${PORT}.log
+    exit 0
+fi
+
+if [ "$1" == "tailall" ]
+then
+    tail -f *.log
+    exit 0
+fi
+
+if [ "$1" == "call" ]
+then
+    while [ $((PORT < ENDPORT)) != "0" ]; do
+        PORT=$((PORT+1))
+        ${REDIS_PATH}/redis-cli -p $PORT $2 $3 $4 $5 $6 $7 $8 $9
```

Comment:
Why not use `shift` and then `$@`?

Reply:
This is copied from Redis, I had the same thought actually but ended up leaving it as is. I'll do that to get rid of the arguments count limit.